### PR TITLE
Triangle Strip & Fan

### PR DESCRIPTION
**Pull request by benaadams**
Follow up for  #6849

### CODE REVIEW

File: src/objects/Mesh.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * @author jonobr1 / http://jonobr1.com/
  */
 
-THREE.Mesh = function ( geometry, material ) {
+THREE.Mesh = function ( geometry, material, mode ) {
```
Comment (mrdoob) [inline src/objects/Mesh.js]: I wouldn't add `mode` here. I would prefer to keep constructors simple and use this sort of API instead:

```javascript
new THREE.Mesh( geometry, material ).setMode( THREE.TriangleFan )
```

**Comment by benaadams** [comment]
More like so?

**Comment by mrdoob** [comment]
Yep! One more thing though...

Wouldn't it be better if it was `.drawMode` and `.setDrawMode`. Also, maybe... `+THREE.TrianglePiecesDrawMode`, etc

**Comment by benaadams** [comment]
Done, think of the bytes! ;-)

**Comment by mrdoob** [comment]
Haha, I'm giving priority to future refactorings 😉

File: src/renderers/WebGLRenderer.js [hunk 2]
```diff
@@ -846,8 +846,22 @@ THREE.WebGLRenderer = function ( parameters ) {
 				renderer.setMode( _gl.LINES );
 
 			} else {
+				
+				switch ( object.mode ) {
 
-				renderer.setMode( _gl.TRIANGLES );
+					case THREE.TrianglePieces:
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: The constants need to be updated.

**Comment by mrdoob** [comment]
Thinking about bytes (and cleanliness)...

Maybe someday we could consider something like:

```javascript
THREE.DRAW_MODE = {
    TRIANGLE: {
        PIECES: 0, STRIP: 1, FAN: 2
    }
};
```

File: src/renderers/WebGLRenderer.js [hunk 3]
```diff
@@ -846,8 +846,22 @@ THREE.WebGLRenderer = function ( parameters ) {
 				renderer.setMode( _gl.LINES );
 
 			} else {
+				
+				switch ( object.mode ) {
 
-				renderer.setMode( _gl.TRIANGLES );
+					case THREE.TrianglePiecesDrawMode:
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: Last one... Wouldn't it be better `THREE.TrianglesDrawMode` instead of `THREE.TrianglePiecesDrawMode`?

**Comment by benaadams** [comment]
> Maybe someday we could consider...

Like it, but with less CAPS ;-)

File: src/renderers/WebGLRenderer.js (hunk 3 above)
Comment (benaadams) [inline src/renderers/WebGLRenderer.js]: Yep

**Comment by mrdoob** [comment]
Thanks!

**Comment by barneygale** [comment]
How can I use this?

**Pull request merged**